test/t/conftest.py

```diff
@@ -186,29 +186,39 @@ def partialize(
 def bash(request) -> pexpect.spawn:
 
     logfile = None
-    if os.environ.get("BASHCOMP_TEST_LOGFILE"):
-        logfile = open(os.environ["BASHCOMP_TEST_LOGFILE"], "w")
-    elif os.environ.get("CI"):
-        logfile = sys.stdout
-    testdir = os.path.abspath(
-        os.path.join(os.path.dirname(__file__), os.pardir)
-    )
-    env = os.environ.copy()
-    env.update(
-        dict(
-            SRCDIR=testdir,  # TODO needed at least by bashrc
-            SRCDIRABS=testdir,
-            PS1=PS1,
-            INPUTRC="%s/config/inputrc" % testdir,
-            TERM="dumb",
-            LC_COLLATE="C",  # to match Python's default locale unaware sort
-            HISTFILE="/dev/null",  # to leave user's history file alone
-        )
-    )
-
+    histfile = None
     tmpdir = None
     bash = None
     try:
+
+        if os.environ.get("BASHCOMP_TEST_LOGFILE"):
+            logfile = open(os.environ["BASHCOMP_TEST_LOGFILE"], "w")
+        elif os.environ.get("CI"):
+            logfile = sys.stdout
+
+        testdir = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), os.pardir)
+        )
+
+        # create temporary file for HISTFILE
+        histfile = tempfile.NamedTemporaryFile(
+            prefix="bash-completion-test_", delete=False
+        )
```

Comment:
We could move the `try:` here before `histfile.close()`, as the above lines do not create any resources we could/should clean up if they fail -- better to keep `try` blocks as focused as possible. But no biggie, leaving up to you to decide.

Reply:
(...or maybe move it back to where it was, I'm not sure we can expect to be able to remove a tempfile that has failed to `close()`, but still not a biggie, feel free to leave as is if you feel it's better this way.)